Packages/Tracking Preview/PhysicsHands/Runtime/PhysicsHand.cs

```diff
@@ -671,6 +694,60 @@ public bool IsObjectInBoneRadius(Rigidbody rigid, PhysicsBone bone, float radius
             return false;
         }
 
+        /// <summary>
+        /// Have the bones been forced beyond acceptable rotation amounts by external forces?
+        /// </summary>
+        private bool AreBonesRotatedBeyondThreshold(float eulerThreshold = 20f)
+        {
+            if (IsGrasping)
+                return false;
+
+            int boneArrayIndex = -1;
+            ArticulationBody body = null;
+
+            for (int fingerIndex = 0; fingerIndex < Hand.FINGERS; fingerIndex++)
+            {
+                for (int jointIndex = 1; jointIndex < Hand.BONES; jointIndex++)
+                {
+                    boneArrayIndex = fingerIndex * Hand.BONES + jointIndex;
+                    
+                    // Skip finger if a bone's contacting
+                    if (_physicsHand.jointBones[boneArrayIndex].IsContacting)
+                    {
+                        break;
+                    }
+
+                    body = _physicsHand.jointBodies[boneArrayIndex];
+                    float angle = Mathf.Repeat(body.transform.localRotation.eulerAngles.x + 180, 360) - 180;
+                    if(angle < body.xDrive.lowerLimit - eulerThreshold)
+                    {
+                        return true;
+                    }
+
+                    if(angle > body.xDrive.upperLimit + eulerThreshold)
+                    {
+                        return true;
+                    }
+
+                    // If the bone's meant to be pretty flat
+                    float delta = Mathf.DeltaAngle(angle, body.xDrive.target);
+                    if(Mathf.Abs(body.xDrive.target) < eulerThreshold / 2f && delta > eulerThreshold)
+                    {
+                        _physicsHand.overRotationCount[boneArrayIndex]++;
+                        if (_physicsHand.overRotationCount[boneArrayIndex] > 10)
```

Comment:
What's with counting 10 joints here before we think they are over-rotated?

Reply:
If you have it done instantly, the bones can be flat and just moving to their correct rotation which would cause this to fire. Counting to 10 gives it 10 frames for the physics simulation to catch up if it's ok. If it's bad then that 10 limit will be hit fairly quickly.

Reply:
im doing some local file changes so i'll add this as comments :)

Reply:
Thanks both.